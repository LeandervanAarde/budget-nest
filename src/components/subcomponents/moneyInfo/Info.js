import React from 'react';
import { Col } from 'react-bootstrap';
const Info = (props) => {
    return (
       <Col md={3} className="infoBlock" id={props.id}>
           <h4 className='text-center'>{props.heading}</h4>
           <p className='text-center' id={props.id2}>{props.content}</p>
           {props.extra}
       </Col>
    );
};

export default Info;