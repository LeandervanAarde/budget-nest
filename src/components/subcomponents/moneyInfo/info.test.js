
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Info from "./Info";
import TestRenderer from 'react-test-renderer';
import { act, create } from "react-test-renderer";

describe("Test the household component", () =>{
    test("Create Rendered component", () => {

        const testRender = TestRenderer.create(
           <Info heading={"Some heading that is reusable"} content={"R 14000.00"}/>
        );
        const testInstance = testRender.root;
        expect(testInstance.findByType(Info).props.content).toBe("R 14000.00");
    });

    test("matches snap", async () => {
        let root;
        await act(() => {
            root = create(<Info heading={"Some heading that is reusable"} content={"R 14000.00"}/>);
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            root.update(<Info heading={"Some heading that is reusable"} content={"R 14000.00"}/>);
            expect(root.toJSON()).toMatchSnapshot();
        })
        console.log(root.toTree)
    });
});