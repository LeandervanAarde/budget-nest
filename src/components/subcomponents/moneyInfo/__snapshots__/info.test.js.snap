// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the household component matches snap 1`] = `
<div
  className="infoBlock col-md-3"
>
  <h4
    className="text-center"
  >
    Some heading that is reusable
  </h4>
  <p
    className="text-center"
  >
    R 14000.00
  </p>
</div>
`;

exports[`Test the household component matches snap 2`] = `
<div
  className="infoBlock col-md-3"
>
  <h4
    className="text-center"
  >
    Some heading that is reusable
  </h4>
  <p
    className="text-center"
  >
    R 14000.00
  </p>
</div>
`;
